### PR TITLE
Bunch transactions response [ECR-1029]

### PR DESCRIPTION
**Pull request by SerhiiOryshych**

### CODE REVIEW

File: exonum/src/node/consensus.rs [hunk 1]
```diff
@@ -589,6 +590,36 @@ impl NodeHandler {
...
+            );
+            return;
+        }
+
+        if !msg.verify_signature(msg.from()) {
+            error!("Received request with incorrect signature, msg={:?}", msg);
+            return;
+        }
+
+        for tx in msg.transactions() {
```
Comment (slowli) [inline exonum/src/node/consensus.rs]: Do I understand correctly that this code clones all raw messages? Doesn't look very good, tbh.

File: exonum/src/messages/mod.rs [hunk 2]
```diff
@@ -55,6 +55,8 @@ pub enum Any {
     Request(RequestMessage),
     /// Transaction.
     Transaction(RawTransaction),
+    /// A bunch of transactions.
+    TransactionsBunch(TransactionsResponse),
```
Comment (slowli) [inline exonum/src/messages/mod.rs]: `Bunch` intuitively seems wrong to me; `TransactionsBundle` or `TransactionsBatch` both sound better. @Maria-Nosyk?

Comment (Maria-Nosyk) [reply exonum/src/messages/mod.rs]: bunch isn't absolutely wrong but has a slightly different connotation, besides batch is more common for us as I could notice from our docs

**Comment by alekseysidorov** [comment]
What about the problem with the messages size limit? Single transaction can fit it, but a several in sum can be bigger than size limit.

File: exonum/src/messages/mod.rs (hunk 2 above)
Comment (DarkEld3r) [reply exonum/src/messages/mod.rs]: Personally, I prefer to use "batch" term here.

**Comment by DarkEld3r** [comment]
@alekseysidorov 
Nice catch, but we already have this problem with `BlockResponse` message. I suppose we can deal with it separately.

File: exonum/src/node/consensus.rs (hunk 1 above)
Comment (SerhiiOryshych) [reply exonum/src/node/consensus.rs]: Yes, you are right. How do you think, how should this be done?

**Comment by SerhiiOryshych** [comment]
@alekseysidorov 
I'm trying to verify this here: https://github.com/exonum/exonum/pull/583/files#diff-dece0e59f8e786debd5d53b790fcccfcR87.

File: exonum/src/messages/protocol.rs [hunk 3]
```diff
@@ -63,6 +65,10 @@ pub const PEERS_REQUEST_MESSAGE_ID: u16 = PeersRequest::MESSAGE_ID;
 /// `BlockRequest` message id.
 pub const BLOCK_REQUEST_MESSAGE_ID: u16 = BlockRequest::MESSAGE_ID;
 
+/// The size of the part of the message intended only for transactions.
+pub const TRANSACTIONS_RESPONSE_UNOCCUPIED_MESSAGE_SIZE: u32 =
```
Comment (vldm) [inline exonum/src/messages/protocol.rs]: Can you rewrite it as local variable, and use real `MAX_MESSAGE_LEN` config?

File: exonum/src/node/consensus.rs [hunk 4]
```diff
@@ -589,6 +590,36 @@ impl NodeHandler {
         }
     }
 
+    /// Handles raw transactions.
+    #[cfg_attr(feature = "flame_profile", flame)]
+    pub fn handle_txs(&mut self, msg: &TransactionsResponse) {
+        if msg.to() != self.state.consensus_public_key() {
+            error!(
+                "Received request that intended for another peer, to={}, from={}",
```
Comment (vldm) [inline exonum/src/node/consensus.rs]: As i understand, this is not request.

File: exonum/src/node/consensus.rs [hunk 5]
```diff
@@ -589,6 +590,36 @@ impl NodeHandler {
...
+                msg.to().to_hex(),
+                msg.from().to_hex()
+            );
+            return;
+        }
+
+        if !self.state.whitelist().allow(msg.from()) {
```
Comment (vldm) [inline exonum/src/node/consensus.rs]: Probably we should ignore "batches" that was not requested perviously.

File: exonum/src/node/requests.rs [hunk 6]
```diff
@@ -12,9 +12,13 @@
...
 // limitations under the License.
 
+use std::mem::replace;
+
 use messages::{BlockRequest, BlockResponse, Message, PrevotesRequest, ProposeRequest,
-               RequestMessage, TransactionsRequest};
+               RequestMessage, TransactionsRequest, TransactionsResponse};
 use blockchain::Schema;
+use crypto::{PUBLIC_KEY_LENGTH, SIGNATURE_LENGTH};
+use messages::HEADER_LENGTH;
```
Comment (vitvakatu) [inline exonum/src/node/requests.rs]: nitpick: what about grouping this import with that one above?

**Pull request merged**